AmazonChimeSDKDemo/AmazonChimeSDKDemo/CaptionsModel.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  CaptionsModel.swift
+//  AmazonChimeSDKDemo
+//
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0
+//
+
+import AmazonChimeSDK
+import UIKit
+
+class CaptionsModel: NSObject {
+    var captions = [Caption]()
+    var captionIndices = [String: Int]()    // map resultId to index in captions array
+    var refreshCaptionsTableHandler: (() -> Void)?
+
+    // triggered on main thread, no synchronization needed
+    public func addTranscriptEvent(transcriptEvent: TranscriptEvent) {
+        if let status = transcriptEvent as? TranscriptionStatus {
+            let formattedEventTime = TimeStampConversion.formatTimestamp(timestamp: status.eventTimeMs)
+            let content = "Live transcription \(status.type) at \(formattedEventTime) in \(status.transcriptionRegion) with configuration: \(status.transcriptionConfiguration)"
+            let caption = Caption(speakerName: "",
+                                  isPartial: false,
+                                  content: content)
+            captions.append(caption)
+        } else if let transcript = transcriptEvent as? Transcript {
+            transcript.results.forEach { result in
+                guard let alternative = result.alternatives.first else {
+                    return
+                }
+
+                // for simplicity and demo purposes, assume each result only contains transcripts from
+                // the same speaker, which matches our observation with current transcription service behavior.
+                // More complicated UI logic can be achieved by iterating through each item
+                var speakerName: String
+                if let firstItem = alternative.items.first {
+                    speakerName = RosterModel.convertAttendeeName(from: firstItem.attendee)
+                } else {
+                    speakerName = ""
```

Comment:
Maybe also add a debug log about this unknown attendee?

Reply:
Theoretically this should never happen. But if it does, it will be visible on the UI though (transcript from an empty speaker UILabel), not sure if adding a debug log can add any value here.

Reply:
At least there's an attendee info to lookup in the server I guess?

Reply:
I see what you mean. When it happens the attendeeId is likely going to be empty as well but I can add the log statement here.